src/fancy-tools/scaleOverlay.js

```diff
@@ -0,0 +1,242 @@
+
+    const startPoint = {
+      x: imageAttributes.verticalLine.start.x,
+      y: imageAttributes.verticalLine.start.y + i * imageAttributes.verticalMinorTick
+    };
+
+    const endPoint = {
+      x: 0,
+      y: imageAttributes.verticalLine.start.y + i * imageAttributes.verticalMinorTick
+    };
+
+    if (i % 5 === 0) {
+
+      endPoint.x = imageAttributes.verticalLine.start.x - imageAttributes.majorTickLength;
+    } else {
+
+      endPoint.x = imageAttributes.verticalLine.start.x - imageAttributes.minorTickLength;
+    }
+
+    drawLine(context, startPoint, endPoint);
+
+    i++;
+  }
+};
+
+const drawHorizontalScalebarIntervals = (context, imageAttributes) => {
+  let i = 0;
+
+  while (imageAttributes.horizontalLine.start.x + i * imageAttributes.horizontalMinorTick <= imageAttributes.hscaleBounds.bottomRight.x) {
+
+    const startPoint = {
+      x: imageAttributes.horizontalLine.start.x + i * imageAttributes.horizontalMinorTick,
+      y: imageAttributes.horizontalLine.start.y
+    };
+
+    const endPoint = {
+      x: imageAttributes.horizontalLine.start.x + i * imageAttributes.horizontalMinorTick,
+      y: 0
+    };
+
+    if (i % 5 === 0) {
+      endPoint.y = imageAttributes.horizontalLine.start.y - imageAttributes.majorTickLength;
+    } else {
+      endPoint.y = imageAttributes.horizontalLine.start.y - imageAttributes.minorTickLength;
+    }
+
+    drawLine(context, startPoint, endPoint);
+
+    i++;
+  }
+};
+
+function drawVerticalScalebar (context, imageAttributes) {
+  const startPoint = {
+    x: imageAttributes.verticalLine.start.x,
+    y: imageAttributes.verticalLine.start.y
+  };
+  const endPoint = {
+    x: imageAttributes.verticalLine.end.x,
+    y: imageAttributes.verticalLine.end.y
+  };
+
+  const { color, lineWidth } = imageAttributes;
+
+  path(context, { color,
+    lineWidth }, (context) => {
+    drawLine(context, startPoint, endPoint);
+    drawVerticalScalebarIntervals(context, imageAttributes);
+  });
+}
+
+function drawHorizontalScalebar (context, imageAttributes) {
+  const startPoint = {
+    x: imageAttributes.horizontalLine.start.x,
+    y: imageAttributes.horizontalLine.start.y
+  };
+  const endPoint = {
+    x: imageAttributes.horizontalLine.end.x,
+    y: imageAttributes.horizontalLine.end.y
+  };
+
+  drawLine(context, startPoint, endPoint);
+  drawHorizontalScalebarIntervals(context, imageAttributes);
+}
+
+export default class extends baseTool {
+  constructor (name) {
+    super({
+      name: name || 'scaleOverlay',
+      configuration: {
+        color: 'white',
+        lineWidth: 2,
+        shadowColor: 'black',
+        shadowBlur: 4,
+        minorTickLength: 12.5,
+        majorTickLength: 25
+      }
+    });
+  }
+
+  enabledCallback (element) {
+    external.cornerstone.updateImage(element);
+  }
+
+  disabledCallback (element) {
+    external.cornerstone.updateImage(element);
+  }
```

Comment:
`enabledCallback` and `disabledCallback` look good 👍 

each `setToolMode` method calls a callback with the following formula:
- `tool['${mode}Callback'](element, options);`